django/db/backends/mysql/operations.py

```diff
@@ -258,3 +259,13 @@ def subtract_temporals(self, internal_type, lhs, rhs):
             ) % {'lhs': lhs_sql, 'rhs': rhs_sql}, lhs_params * 2 + rhs_params * 2
         else:
             return "TIMESTAMPDIFF(MICROSECOND, %s, %s)" % (rhs_sql, lhs_sql), rhs_params + lhs_params
+
+    def explain_query_prefix(self, output_format=None, verbose=False):
+        prefix = super().explain_query_prefix(output_format, verbose)
+        # With mysql EXTENDED and FORMAT=json cannot be used together.
+        if output_format:
+            prefix += ' FORMAT=%s ' % output_format
+        elif verbose:
+            # EXTENDED format is deprecated in 5.7 and above.
+            prefix += ' EXTENDED '
+        return prefix
```

Comment:
`EXTENDED` is removed in MySQL 8.0 as it has become part of the default output as of 5.7. This needs to be changed to only be added for MySQL < 5.7

`PARTITIONS` is also missing but the same situation as `EXTENDED` above also applies. It is also mutually exclusive with `EXTENDED` and `FORMAT`. It is probably not worth adding the capability to get the `PARTITIONS` output only for 5.6

According to the [documentation](https://dev.mysql.com/doc/refman/5.6/en/explain.html#explain-execution-plan) the JSON format already includes the extra information provided by `EXTENDED` and `PARTITIONS`.

I suggest changing this method to something like the following (which includes the text format option translation mentioned in another comment):

```python
def explain_query_prefix(self, output_format=None, verbose=False):
    prefix = super().explain_query_prefix(output_format, verbose)
    if output_format:
        if output_format.upper() == 'TEXT':
            output_format = 'TRADITIONAL'
        prefix += ' FORMAT=%s ' % output_format
    if self.connection.mysql_version < (5, 7) and verbose and output_format is None:
        # EXTENDED and FORMAT are mutually exclusive options.
        # EXTENDED is deprecated (and not required) in 5.7 and removed in 8.0
        prefix += ' EXTENDED '
    return prefix
```

This makes it easier to remove the cruft when MySQL 5.6 is no longer supported.

I'm not sure whether it is worth having the default for `verbose` be `None` thus turning this flag tri-state. This would would allow for raising an exception when `verbose` is explicitly set to `False` for MySQL >= 5.7, but I'll defer to someone else's better judgement.

Reply:
Thank you for this, and for the code snippet. I'll work on adding support for this.

As I understand it mysql's explain output is very different from other databases, the default output is in the form of a single tuple, and the column names are very important in the output. Currently there isn't a way to return the column names from queries, which I think makes this a little bit useless on mysql currently. Would you agree, and would you know any workarounds? I'm a bit loathe to add support for returning column names from some of the internals just for this specific case.

Reply:
Ignore that, you can get the column names, I was just being dense 👍

Reply:
I think it is worth just spewing it out as a complete text blob. Otherwise we would be adding specialised parsing which is not worth it for a textual output.

Reply:
Also, could you add in the handling for `EXTENDED` as I had in my comment above? This will ensure that Django will output the same, regardless of the version of MySQL.

Reply:
Done 👍